lib/cache/senders/impressions_sender.rb

```diff
@@ -18,20 +18,30 @@ def call
           if ENV['SPLITCLIENT_ENV'] == 'test'
             post_impressions
           else
-            Thread.new do
-              @config.logger.info('Starting impressions service')
+            impressions_thread
 
-              loop do
-                post_impressions
-
-                sleep(randomize_interval(@config.impressions_refresh_rate))
+            if defined?(PhusionPassenger)
+              PhusionPassenger.on_event(:starting_worker_process) do |forked|
```

Comment:
@maksimf did you test this yourself? any other consequence of resurrenting the threads?